src/support/small_set.h

```diff
@@ -61,16 +82,24 @@ struct UnorderedFixedStorage : public FixedStorageBase<T, N> {
 
 template<typename T, size_t N>
 struct OrderedFixedStorage : public FixedStorageBase<T, N> {
-  void insert(const T& x) {
-    assert(this->used < N);
+  using InsertResult = typename FixedStorageBase<T, N>::InsertResult;
 
+  InsertResult insert(const T& x) {
     // Find the insertion point |i| where x should be placed.
     size_t i = 0;
     while (i < this->used && this->storage[i] <= x) {
+      if (this->storage[i] == x) {
+        return InsertResult::AlreadyExists;
+      }
```

Comment:
It would be better to change the loop to use `<` instead of `<=` and to move this check to be after the loop.